r-package/grf/R/causal_forest.R

```diff
@@ -144,24 +147,37 @@ causal_forest <- function(X, Y, W,
 
     reduced.form.weight <- 0
 
-    if (is.null(Y.hat)) {
+    if (is.null(Y.hat) && !boosting) {
```

Comment:
There's already a lot going on in `causal_forest`, and I don't know if we should be adding more to it right now.

Is there a reason for not creating a separate function `boosted_causal_forest`, just like we have `boosted_regression_forest`?